server/src/main/java/io/crate/protocols/postgres/PostgresWireProtocol.java

```diff
@@ -867,6 +900,47 @@ private ByteBuf nullOrBuffer(ByteBuf buffer, State nextState) {
             state = nextState;
             return buffer.readBytes(msgLength);
         }
+
+        private State nextState(int protocol) {
+            switch (protocol) {
+                case 80877102:
+                    return State.CANCEL_BODY;
+                default:
+                    return State.STARTUP_BODY;
+            }
+        }
     }
 
+    private static class KeyData {
+        private final int pid;
+        private final int secretKey;
+
+        public KeyData() {
+            var random = ThreadLocalRandom.current();
+            this.pid = random.nextInt();
```

Comment:
I'd go with a incrementing number. `random.nextInt()` could return the same number on successive calls - leading to conflicts.